src/DefaultBuilder/test/Microsoft.AspNetCore.Tests/WebApplicationTests.cs

```diff
@@ -1976,6 +1981,75 @@ public void CanObserveDefaultServicesInServiceCollection()
         Assert.Contains(builder.Services, service => service.ServiceType == typeof(ILogger<>));
     }
 
+    [Fact]
+    public async Task RegisterAuthMiddlewaresCorrectly()
+    {
+        var helloEndpointCalled = false;
+        var customMiddlewareExecuted = false;
+        var username = "foobar";
+
+        var builder = WebApplication.CreateBuilder();
+        builder.Services.AddAuthenticationCore(o =>
+        {
+            o.AddScheme<UberHandler>("testSchemeName", "testDisplayName");
+            o.DefaultScheme = "testSchemeName";
+        });
+        builder.WebHost.UseTestServer();
+        builder.Authentication.AddJwtBearer();
+        await using var app = builder.Build();
+
+        app.Use(next =>
+        {
+            return async context =>
+            {
+                // IAuthenticationFeature is added by the authentication middleware
+                // during invocation. This middleware should run after authentication
+                // and be able to access the feature.
+                var authFeature = context.Features.Get<IAuthenticationFeature>();
+                Assert.NotNull(authFeature);
+                customMiddlewareExecuted = true;
+                Assert.Equal("foobar", context.User.Identity.Name);
+                await next(context);
+            };
+        });
+
+        app.MapGet("/hello", () => helloEndpointCalled = true).AllowAnonymous();
```

Comment:
Can you also verify the user name is set here too?

Reply:
Done.